doc/plugin-tutorial.md

```diff
@@ -675,35 +655,80 @@ function formatSiacoin(hastings) {
-	});
-	callAPI('/gateway/status', function(result) {
-		document.getElementById('peers').innerHTML = 'Peers: ' + result.Peers.length;
-	});
-	callAPI('/consensus/status', function(result) {
-		document.getElementById('block-height').innerHTML = 'Block Height: ' + result.Height;
-	});
-	listening = true;
-}
-
-// Called upon showing
-function init() {
+// Called by the UI upon showing
+function start() {
 	// DEVTOOL: uncomment to bring up devtools on plugin view
 	// IPC.sendToHost('devtools');
 	
-	// Ensure precision
-	BigNumber.config({ DECIMAL_PLACES: 24 });
-	BigNumber.config({ EXPONENTIAL_AT: 1e+9 });
-	
-	// Call the API regularly to update page
-	updating = setInterval(update, 1000);
+	// Call the API
+	update();
 }
 
-// Called upon transitioning away from this view
-function kill() {
-	clearInterval(updating);
+// Called by the UI upon transitioning away from this view
+function stop() {
+	clearTimeout(updating);
+}
+
+// Define IPC listeners and update DOM per call
+IPC.on('balance-update', function(err, result) {
+	updateField(err, 'Balance: ', formatSiacoin(result.Balance), 'balance');
+});
+IPC.on('peers-update', function(err, result) {
+	updateField(err, 'Peers: ', result.Peers.length, 'peers');
+});
+IPC.on('height-update', function(err, result) {
+	updateField(err, 'Block Height: ', result.Height, 'height');
+});
+```
+
+### Bonus: Tooltips and Notifications
+
+The UI has a notification and tooltip system that can also be utilized through
+IPC. Notifications are easy enough as demonstrated through this code snippet
+taken from Wallet's js file:
+
+```js
+IPC.on('coin-sent', function(err, result) {
+	if (err) {
+		console.error(err);
+		IPC.sendToHost('notify', 'Transaction errored!', 'error');
+		return;
+	}
+	IPC.sendToHost('notify',  'Transaction sent!', 'sent');
+	document.getElementById('transaction-amount').value = '';
+	document.getElementById('confirm').classList.add('hidden');
+});
+```
+
+In the wallet view, we can trigger this notification and another, making them
+appear as such: ![If only we sent to an actual addresses]
+(/doc/assets/wallet-notifications.png)
+
+In a similar, but more unwieldy manner, one can send a tooltip IPC message to
+show atop an element as demonstrated again by wallet:
+
+```js
+// Ask UI to show tooltip bubble
+function tooltip(message, element) {
+	var rect = element.getBoundingClientRect();
+	IPC.sendToHost('tooltip', message, {
+		top: rect.top,
+		bottom: rect.bottom,
+		left: rect.left,
+		right: rect.right,
+		height: rect.height,
+		width: rect.width,
+		length: rect.length,
+	});
 }
+
+// Give the buttons interactivity
+document.getElementById('create-address').onclick = function() {
+	tooltip('Creating...', this);
+	IPC.sendToHost('api-call', '/wallet/address', 'new-address');
+};
 ```
+
+Which will produce this tooltip on click appear as such:
+
+![It's the details that make the best UI] (/doc/assets/wallet-tooltip.png)
```

Comment:
this image (and the one above) are showing up as links in the preview. There isn't usually a space between the `]` and `(`, maybe that's the problem. Otherwise it could just be a diff rendering quirk.